nixos/modules/system/boot/loader/systemd-boot/systemd-boot-builder.py

```diff
@@ -148,12 +152,11 @@ def bootspec_from_json(bootspec_json: dict[str, Any]) -> BootSpec:
 
 
 def copy_from_file(file: str, dry_run: bool = False) -> str:
-    store_file_path = os.path.realpath(file)
-    suffix = os.path.basename(store_file_path)
-    store_dir = os.path.basename(os.path.dirname(store_file_path))
-    efi_file_path = f"{NIXOS_DIR}/{store_dir}-{suffix}.efi"
+    store_file_path = Path(os.path.realpath(file))
+    store_subdir = next(iter(set(store_dir.parts) ^ set(store_file_path.parts)))
```

Comment:
After a bit of discussion on matrix, seems like this line needs some attention, as sets are not ordered.

Reply:
Changed to use `os.path.relpath` instead, which should look much simpler